firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/ComponentRegistrarTest.kt

```diff
@@ -97,4 +98,42 @@ class ComponentRegistrarTest {
         val actual = waitFor(docRefActual.get()).toObject<TestObj>()
         assertThat(actual).isEqualTo(expected)
     }
+
+    @Serializable
+    private data class DocumentIdObj(@DocumentId val docId: String = "should be null in Java")
+
+    @Test
+    fun docId_annotation_should_be_shared_between_java_and_kotlin() {
+        // Kotlin can see this @DocumentId annotation
+        val annotations = DocumentIdObj.serializer().descriptor.getElementAnnotations(0)
+        val annotation = annotations[0]
+        annotation::class.java isAssignableTo DocumentId::class.java
+
+        // Java POJO should also see this annotation and use it to skip the annotated field
+        val docRef = testCollection("pojo").document("456")
+        docRef.withoutCustomMappers { set(DocumentIdObj()) }
+        val pojoMap = waitFor(docRef.get()).data
+        pojoMap shouldBe emptyMap
```

Comment:
I have changed the Java custom mapper to make Java code able to capture annotations on Property. This is the test.